### PR TITLE
Feature/second milestone

### PR DESCRIPTION
**Pull request by codesalley**

### CODE REVIEW

File: app/views/friendships/_all_friends.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<% if @allfriends.size < 1%>
+<p class="border">You have no friends yet</p>
+<% else %> 
+<p class="border">All friends</p>
+<% @allfriends.each do |friend| %>
+
+<p class="">
+    <%=link_to friend.email, user_path(friend) %></p>
+<% end %> <% end %>
```
Comment (lekegitrepo) [inline app/views/friendships/_all_friends.html.erb]: - Remove logic from the view

File: app/views/friendships/_received_requests.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+<p class=""><%= friend.email %></p>
+<div class="options">
+<%= accept_request(friend)%>
+<%= reject_request(friend) %>
+</div>
+
+<% end %> 
+
+</div>
+<% end %>
```
Comment (lekegitrepo) [inline app/views/friendships/_received_requests.html.erb]: - Remove logic from the view

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -19,6 +21,7 @@
         </div>
       </div>
       <% if current_user %>
+      <%= link_to 'Friend requests', friendships_path, class: 'btn btn-outline-danger'  %>
         <%= link_to 'Sign out', destroy_user_session_path, method: :delete %>
       <% else %>
         <%= link_to 'Sign in', user_session_path %>
```
Comment (lekegitrepo) [inline app/views/layouts/application.html.erb]: - Remove logic from the view

File: app/views/users/_user.html.erb [hunk 4]
```diff
@@ -1,6 +1,10 @@
...
+<li class="my-2 w-100">
   Name: <%= user.name %>
-  <span class="profile-link">
-    <%= link_to 'See Profile',  user_path(user), class: 'profile-link' %>
+  <span class="profile-link my-2">
+    <%= link_to 'See Profile',  user_path(user), class: 'profile-link btn btn-outline-success mx-1 py-2' %>
+    <%= show_btn(user) %>
+    <%if  available_users?(user) %>
+     <%= add_request(user)%>    
+    <% end %>
```
Comment (lekegitrepo) [inline app/views/users/_user.html.erb]: - Remove logic from the view

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @juxsalley :clap: :clap:
#### You have done a good job :+1: the models are well implemented :muscle:
#### However, some requirements are missing/not implemented correctly. Check my comments below :point_down:

#### 1) App Behaviour:
 - There should be a button to invite a user to friendship on the user single show page
 - Posts timeline should include posts written by logged-in user and their friends

#### 2) Test:
 - Add integration tests

#### 3) View & Logic:
 - There should be NO application logic in the views use the helper method instead. Check these links for details [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code), [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers),


#### _Reviewed by Temitayo Adeleke_

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Nice work so far. 👍 
- In a clean install of your project I was able to set up two accounts, and send a friend request from account No. 1 to account No. 2. Good work there.
- Good work with the changes requested by the previous TSE.

## However, there are some changes required 👀 🧰 🔨 🔧 
- When I tried logging into account No. 2 to see the friend request, I was not able to find a button anywhere that would allow me to accept it. The only button that appears on account No. 1's username after receiving their friend request, is the "Request friendship" button. There should be a different button that says something like "Accept pending friendship". Please fix this so that I can accept friendship requests.
### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Review by mke2111 — APPROVED** [review]
# STATUS: APPROVED 🥇  🟡 

## Hey Happy Coder, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval.

- [ ] [OPTIONAL] It's best practice to exclude any kind of logic in any of the files in the views folder. You have a few logic statements in the files there. Transfer them to the helpers folder and create methods for them.

### You may now merge to the develop branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 

## Reach me on slack for a chit-chat 🆘  Roy Mukuye

**Pull request merged**